Tasks/Common/Deployment/TelemetryHelper/TelemetryHelper.psm1

```diff
@@ -1,38 +1,65 @@
 # Telemetry Codes
-$telemetryCodes = 
+$telemetryCodes =
 @{
-  "Input_Validation" = "Input_Validation_Error";
-  "Task_InternalError" = "Task_Internal_Error";
-  "DTLSDK_Error" = "Dtl_Sdk_Error";  
- }
+    "Input_Validation"   = "Input_Validation_Error";
+    "Task_InternalError" = "Task_Internal_Error";
+    "DTLSDK_Error"       = "Dtl_Sdk_Error";
+}
 
- # Telemetry Write Method
+# Telemetry Write Method
 function Write-Telemetry
 {
- [CmdletBinding()]
-  param(
-    [Parameter(Mandatory=$True,Position=1)]
-    [string]$codeKey,
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory = $True, Position = 1)]
+        [string]$codeKey,
+
+        [Parameter(Position = 2)]
+        [string]$errorMsg
+    )
+
+    $erroCodeMsg = $telemetryCodes[$codeKey]
+
+    ## If no error is passed mark it as not available
+    if ([string]::IsNullOrEmpty($errorMsg))
+    {
+        $errorMsg = "No error details available"
+    }
+    $erroCode = ('"{0}":{1}' -f $erroCodeMsg, $errorMsg)
+    ## Form errorcode as json string
+    $erroCode = '{' + $erroCode + '}'
```

Comment:
This doesn't ensure json string if char escaping etc is required. Use native ways to genrate the json

Reply:
done.